vendor/github.com/ipfs/go-ipfs/namesys/routing.go

```diff
@@ -230,7 +250,7 @@ func (r *routingResolver) resolveOnce(ctx context.Context, name string) (path.Pa
 		r.cacheSet(name, p, entry)
 		go func() {
 			r.datastore.Put(ds.NewKey(cachePrefix+name), val)
-			r.datastore.Put(ds.NewKey(keyCachePrefix+name), pubkeyBytes)
+			r.datastore.Put(ds.NewKey(keyCachePrefix+hash.B58String()), pubkeyBytes)
 		}()
 		return p, nil
 	}
```

Comment:
This is pushing us away from mainlining vanilla IPFS. Is there no other way?

Reply:
The only other way is to copy this entire package into an internal package and duplicate everything with these few tweaks.